### ISSUE TITLE
bigkyiv.com.ua

### ISSUE THREAD

**Issue by dportvine** [uBlockOrigin/uBOL-home#181]
`https://bigkyiv.com.ua/u-darnyczi-zyavyvsya-svij-spajdermen-video/`

`bigkyiv.com.ua#@#.ad-text` from **[RU AdList](https://github.com/easylist/ruadlist/commit/578da0c6676532a6dd37f6ace4ebff67462056cb)** don't works in complete mode.

<details>
<summary>Screen</summary>

https://github.com/user-attachments/assets/17503669-7bb9-4312-8eaa-37ddb9faceb4
</details>

Google Chrome
uBlock Origin Lite 2024.8.12.902

**Comment by garry-ut99** [comment]
I can reproduce, on both Chromium and Firefox.
 I don't see this filter present in uBOL at all, I've found only this:  
`\rulesets\scripting\generic\default.js(36): ,[13232956,".ad-text"]` 
but it's a generic hiding rule from EasyList: 
https://github.com/easylist/easylist/blob/bc5308e57afa084a7b6a70db1f230e6744fc56d3/easylist/easylist_general_hide.txt#L5696

**Comment by gorhill** [comment]
Exceptions to generic cosmetic filters are not supported in uBOL at the moment.

**Comment by stephenhawk8054** [comment]
@gorhill I wonder if `ghide` works in MV3?

**Comment by garry-ut99** [comment]
Yes it does work.

An example:  on `supercomputingonline.com` after adding `@@||supercomputingonline.com^$ghide,badfilter` an anti-adblock appears, which means the `ghide` is was necessary to bypass anti-adblock.
No anti-adblock appears in uBOL, hence the `ghide` exception does work as expected.

<hr>

> dportvine : `bigkyiv.com.ua#@#.ad-text` from **[RU AdList](https://github.com/easylist/ruadlist/commit/578da0c6676532a6dd37f6ace4ebff67462056cb)** doesn't works in complete mode.

Duplicate of https://github.com/uBlockOrigin/uBOL-home/issues/58.

**Comment by dportvine** [comment]
@dimisa-RUAdList

Can you add `@@||bigkyiv.com.ua^$ghide` for `mv3`?

**Comment by stephenhawk8054** [comment]
I created a section for exceptions to generic cosmetic filters issues: https://github.com/uBlockOrigin/uAssets/blob/master/filters/ubol-filters.txt#L11 since there's a similar issue in uAssets.

**Comment by dimisa-RUAdList** [comment]
Using generichide entails the need to add too many rules specific to the target domain.

That's why I would like to suggest using the style: `display: block !important` instead of generichide, like here: https://github.com/easylist/ruadlist/commit/59abed475da8348792b91abf7f3852b007ff3231

I am also interested in the possibility of making the application of this style automatic for mv3.

**Comment by dportvine** [comment]
> I am also interested in the possibility of making the application of this style automatic for mv3.

@gorhill 

How relevant is it and is it possible to implement such a function?

**Comment by dportvine** [comment]
>[mv3] Improve generic cosmetic filtering

It works for me in the beta version, tested on `https://www.offmoto.com/`

```adblock
##.reklama 
offmoto.com#@#.reklama
```

uBlock Origin Lite 2025.3.8.1350

<details>
<summary>Screen</summary>

![2025-03-14_194113](https://github.com/user-attachments/assets/370a2e1c-7411-42d5-b8f1-b633ae1a5ae6)
</details>

uBlock Origin Lite 2025.3.13.1386-beta

<details>
<summary>Screen</summary>

![2025-03-14_201525](https://github.com/user-attachments/assets/adcd9658-4f4f-46b8-8345-3b905d30cd54)
</details>

**Comment by stephenhawk8054** [comment]
Yeah, looks like it's fixed. I've removed the section of exceptions for generic cosmetic filters: https://github.com/uBlockOrigin/uAssets/commit/d94ac69813d6857739a83d6efe7eae9a0daca52d
